### ISSUE TITLE
InvalidOperationException inside MsgPack (Do not convert System.String MessagePackObject to System.Int64.)

### ISSUE THREAD

**Issue by cspwizard** [msgpack/msgpack-cli#204]
```
Do not convert System.String MessagePackObject to System.Int64.    

at MsgPack.MessagePackObject.ThrowInvalidTypeAs[T](MessagePackObject instance)
   at MsgPack.MessagePackObject.AsInt64()
   at MsgPack.Serialization.DefaultSerializers.NativeDateTimeMessagePackSerializer.UnpackFromCore(Unpacker unpacker)
   at MsgPack.Serialization.MessagePackSerializer`1.UnpackFrom(Unpacker unpacker)
   at MsgPack.Serialization.MessagePackSerializer`1.Unpack(Stream stream)
   at MsgPack.Serialization.MessagePackSerializer`1.UnpackSingleObject(Byte[] buffer)
   at _HashSetSeralizationFailsAndWriteFails() in Test.cs:line 391
```

This issue occurs when I try to deserialize serialized string as `DateTime`. Here is a reproduce code:
```
var date = MessagePackSerializer.Get<DateTime>().UnpackSingleObject(MessagePackSerializer.Get<string>().PackSingleObject("test"))
```
I would expect to get standard `SerializationException` as MsgPack does whet I try to deserialize same serialized string as a `HashSet<string>` instance.

**Comment by yfakariya** [comment]
Thank you, you are right. It should be appropriate type and message.

**Comment by cspwizard** [comment]
Can you give an estimate on new version release date with this issue fix?

**Comment by yfakariya** [comment]
Unfortunately, I'm busy in this month, so it will be next year. Sorry.

**Comment by yfakariya** [comment]
Fixed in 	6b25651
